2_process/src/write_data.R

```diff
@@ -30,3 +30,34 @@ write_df_to_zarr <- function(df, index_cols, out_zarr) {
   
   return(out_zarr)
 }
+
+
+write_dist_matrix_npz <- function(mat_list, out_npz) {
+  #' @title Write an R list of matrices to a numpy npz store
+  #'
+  #' @description This function uses {reticulate} to write an R list of matrices to a numpy data
+  #' store (the file format river-dl currently takes). 
+  #'
+  #' @param mat_list list of matrices. Columns and rows must all be the same order in each matrix
```

Comment:
What do you think of adding a brief comment here that we're expecting a list containing four matrices named "complete," "downstream," "upstream," and "updown?"

Reply:
Thanks! I meant to add that comment. I could not figure out how to get the `keys` automatically recognized within `savez_compressed`. In Python, a named dict is automatically converted to have the same names in the npz file, but that did not happen using reticulate